language-plutus-core/src/Language/PlutusCore/Error.hs

```diff
@@ -1,74 +1,63 @@
-{-# LANGUAGE ConstrainedClassMethods #-}
-{-# LANGUAGE DeriveAnyClass          #-}
-{-# LANGUAGE FlexibleInstances       #-}
-{-# LANGUAGE OverloadedStrings       #-}
-
-module Language.PlutusCore.Error ( Error (..)
-                                 , NormalizationError (..)
-                                 , RenameError (..)
-                                 , TypeError (..)
-                                 , IsError (..)
-                                 ) where
-
-import           Language.PlutusCore.Lexer
+{-# LANGUAGE DeriveAnyClass    #-}
+{-# LANGUAGE FlexibleInstances #-}
+
+module Language.PlutusCore.Error
```

Comment:
I had a lot of pain with errors due to the fact that I first wrote a version of the pretty-printing machinery that tracks configs at the type-level and `TypeError` doesn't really fit into this picture, so I refactored everything and then again and then I chose another approach to pretty-printing, but I think the refactoring is still useful. It moves `PrettyBy` instance to a separate file (and formats the lines there so they're not 330 symbols long) and also moves `ParseError` here from the `Lexer.x` module (it should be heavily refactored: no need to define all this stuff in a *lexing* module). I also threw away some unused stuff from the `IsError` class.